docs/install/docker-security.md

```diff
@@ -0,0 +1,198 @@
+cluster.routing.allocation.disk.threshold_enabled: false
+opendistro_security.audit.config.disabled_rest_categories: NONE
+opendistro_security.audit.config.disabled_transport_categories: NONE
+```
+
+To start the cluster, run `docker-compose up`.
+
+If you encounter any `File /usr/share/elasticsearch/config/elasticsearch.yml has insecure file permissions (should be 0600)` messages, you can use `chmod` on your local machine to set file permissions before running `docker-compose up`. Docker Compose passes files to the container as-is.
+{: .note }
+
+
+## Change passwords for read-only users
+
+After the cluster starts, change the passwords for the read-only user accounts (`admin` and `kibanaserver`). Run `docker ps` to find the `odfe-node1` container ID. Then run:
+
+```
+$ docker exec <container-id> /bin/sh /usr/share/elasticsearch/plugins/opendistro_security/tools/hash.sh -p newpassword
+```
+
+The hash script returns a hashed password (e.g. `$2y$12$SFNvhLHf7MPCpRCq00o/BuU8GMdcD.7BymhT80YHNISBHsfJwhTou`), which you can then copy and paste into `internal_users.yml`. Repeat the process as necessary for all read-only users. Don't worry about the other user accounts; you can change (or delete) them in Kibana.
+
+When you're satisfied, modify `custom-kibana.yml` to include the new `kibanaserver` password. Then restart the cluster using `docker-compose down -v` and `docker-compose up`.
+
+`internal_users.yml` looks like this:
+
+```yml
+# New password applied
+admin:
+  readonly: true
+  hash: $2y$12$SFNvhLHf7MPCpRCq00o/BuU8GMdcD.7BymhT80YHNISBHsfJwhTou
+  roles:
+    - admin
+  attributes:
+    #no dots allowed in attribute names
+    attribute1: value1
+    attribute2: value2
+    attribute3: value3
+
+# Still using default password: logstash
+logstash:
+  hash: $2a$12$u1ShR4l4uBS3Uv59Pa2y5.1uQuZBrZtmNfqB3iM/.jL0XoV9sghS2
+  roles:
+    - logstash
+
+# New password applied
+kibanaserver:
+  readonly: true
+  hash: $2a$12$4AcgAt3xwOWadA5s5blL6ev39OXDNhmOesEoo33eZtrq2N0YrU3H.
+
+# Still using default password: kibanaro
+kibanaro:
+  hash: $2a$12$JJSXNfTowz7Uu5ttXfeYpeYE0arACvcwlPBStB1F.MI7f0U9Z4DGC
+  roles:
+    - kibanauser
+    - readall
+
+# Still using default password: readall
+readall:
+  hash: $2a$12$ae4ycwzwvLtZxwZ82RmiEunBbIPiAmGZduBAjKN0TXdwQFtCwARz2
+  #password is: readall
+  roles:
+    - readall
+
+# Still using default password: snapshotrestore
+snapshotrestore:
+  hash: $2y$12$DpwmetHKwgYnorbgdvORCenv4NAK8cPUg8AI6pxLCuWf/ALc0.v7W
+  roles:
+    - snapshotrestore
+```
+
+
+## Next steps
+
+After the cluster starts, verify the new password:
+
+```bash
+curl -XGET https://localhost:9200 -u admin:admin -k
+Unauthorized
+
+curl -XGET https://localhost:9200 -u admin:newpassword -k
+{
+  ...
+  "tagline" : "You Know, for Search"
+}
+```
+
+Then you can open Kibana at [http://localhost:5601](http://localhost:5601), sign in, and perform additional user management in the **Security** panel.
```

Comment:
Let's document `https` since SSL is setup in the demo configuration. @aetter Have you tested this with https?

Reply:
Huh, this is really interesting. The Docker Compose file I've been using (the one without all these overrides) doesn't start Kibana on HTTPS. Because all our curl calls are HTTPS, I'd never even noticed about Kibana. Do we know how to make that happen?

Reply:
Further testing indicates that you probably can't avoid the Chrome/Firefox "this page is unsafe" warning without a commercial certificate. I'm going to punt on this and keep using Kibana over HTTP.